src/database-layer/db.ts

```diff
@@ -108,6 +108,32 @@ export const engines: {
 	[engine: string]: (connectString: string | object) => Database;
 } = {};
 
+const transactionLockNamespaceMap: Dictionary<number> = {};
+
+export function registerTransactionLockNamespace(
+	namespaceKey: string,
+	namespaceId: number,
```

Comment:
This is an integer type in postgres so at least for that needs to be within -2147483648 to +2147483647 (https://www.postgresql.org/docs/13/datatype-numeric.html), I think for now just checking 0-2147483647 would be fine - we can always expand it later as need be if we for some reason want more